src/brsTypes/components/Group.ts

```diff
@@ -0,0 +1,26 @@
+import { RoSGNode } from "./RoSGNode";
+import { AAMember } from "./RoAssociativeArray";
+
+export class Group extends RoSGNode {
+    readonly builtInFields = [
+        { name: "visible", type: "boolean" },
+        { name: "opacity", type: "float" },
+        { name: "translation", type: "array" },
+        { name: "rotation", type: "float" },
+        { name: "scale", type: "array" },
+        { name: "scaleRotateCenter", type: "array" },
+        { name: "childRenderOrder", type: "string" },
+        { name: "inheritParentTransform", type: "boolean" },
+        { name: "inheritParentOpacity", type: "boolean" },
+        { name: "clippingRect", type: "array" },
+        { name: "renderPass", type: "integer" },
+        { name: "muteAudioGuide", type: "boolean" },
+        { name: "enableRenderTracking", type: "boolean" },
+        { name: "renderTracking", type: "string" },
+    ];
+
+    constructor(members: AAMember[] = []) {
+        super(members, "Group");
+        this.registerFields(this.builtInFields);
```

Comment:
I might be misunderstanding, but won't this overwrite the `members` that `Group` is created with? As far as I can tell, the order of operations is:
1. Call `super(members, "Group")`
2. In the `RoSGNode` constructor, call `this.registerFields` with `builtInFields`
3. In the `RoSGNode` constructor, add all `members` to `this.fields` (intentionally overwriting any of the default `builtInFields` with the `members` that the node was created with).
4. Come back to `Group` constructor.
5. Call `this.registerFields` with `builtInFields`. **At this point, we are overwriting any of the `members` that `Group` passed into `super()` previously.**

Reply:
I think that sounds right, but only in the case where there is a collision between field names would the given value be overwritten. I think you bring up a good point, though, regarding whether `builtInFields` should supersede provided fields a la `members`.

Reply:
Does this not just register fields with the `Group` and not the `RsSGNode`? If the `members` are being pulled from the `Group` then I would not expect them to be overwritten by the `builtInFields`. Roku seems to mostly flatten things into a single scope so the hierarchy becomes important. But it might be worth experimenting with the platform to see what it's doing explicitly.